src/client.cc

```
@@ -92,6 +93,43 @@ struct AlignedBuffer {
 };
 
 #ifdef MINIO_CPP_RDMA
+// The HTTP fallbacks below fill or drain the caller's buffer with ordinary host
+// loads and stores, which fault on a CUDA device pointer. Stage through host
+// memory instead. The two driver-API copies are resolved lazily with dlopen so
+// libminiocpp keeps its "no CUDA link dependency" property: a host without the
+// driver simply reports the copy as unavailable and the caller gets an error
+// rather than a crash.
+struct CudaHostCopy {
```

Cppcheck (2.21.0)

[performance] Prefer prefix ++/-- operators for non-primitive types.

(postfixOperator)